sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -616,6 +616,103 @@ def download_blob(self, offset=None, length=None, **kwargs):
             **kwargs)
         return StorageStreamDownloader(**options)
 
+    def _quick_query_options(self, query_expression,
+                             **kwargs):
+        # type: (str, **Any) -> Dict[str, Any]
+        input_serialization = kwargs.pop('input_serialization', None)
+        output_serialization = kwargs.pop('output_serialization', None)
+        query_request = QueryRequest(expression=query_expression,
+                                     input_serialization=get_quick_query_serialization_info(input_serialization),
+                                     output_serialization=get_quick_query_serialization_info(output_serialization))
+        access_conditions = get_access_conditions(kwargs.pop('lease', None))
+        mod_conditions = get_modify_conditions(kwargs)
+
+        cpk = kwargs.pop('cpk', None)
+        cpk_info = None
+        if cpk:
+            if self.scheme.lower() != 'https':
+                raise ValueError("Customer provided encryption key must be used over HTTPS.")
+            cpk_info = CpkInfo(encryption_key=cpk.key_value, encryption_key_sha256=cpk.key_hash,
+                               encryption_algorithm=cpk.algorithm)
+        options = {
+            'query_request': query_request,
+            'lease_access_conditions': access_conditions,
+            'modified_access_conditions': mod_conditions,
+            'cpk_info': cpk_info,
+            'progress_callback': kwargs.pop('progress_callback', None),
+            'snapshot': self.snapshot,
+            'timeout': kwargs.pop('timeout', None),
+            'cls': return_headers_and_deserialized,
+            'client': self._client,
+            'name': self.blob_name,
+            'container': self.container_name}
+        options.update(kwargs)
+        return options
+
+    @distributed_trace
+    def query(self, query_expression,  # type: str
```

Comment:
I think this should be 
`def query_blobs(self, query, **kwargs)`

Thoughts?

Reply:
the returned items are not blobs, it's the partial content of a single blob(eg. the first column of a csv file)

Reply:
all the platforms are calling it query, so do we want all platforms to rename it?